server/src/test/java/org/elasticsearch/action/bulk/SimulateBulkRequestTests.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.bulk;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class SimulateBulkRequestTests extends ESTestCase {
+
+    public void testSerialization() throws Exception {
+        Map<String, Map<String, Object>> pipelineSubstitutions = getTestPipelineSubstitutions();
+        SimulateBulkRequest simulateBulkRequest = new SimulateBulkRequest();
+        simulateBulkRequest.setPipelineSubstitutions(pipelineSubstitutions);
```

Comment:
Maybe wrap this in a `randomBoolean()` or `frequently()` to test serialization with no substitutions?